lib/gmail/gmail.go

```diff
@@ -364,6 +364,87 @@ func shardForMsgId(id string) int {
 	return int(shard)
 }
 
+// This function return true if the given label parameter
+// is not an extra (decorating) label.
+func isToBeAccounted(label string) bool {
+	// The following build a set of the real mails labels.
+	// Only these mails types are to take into account when
+	// retrieving a full mailbox without filtering by a given
+	// label.
+	type void struct{}
+	var member void
+	toAccountLabels := make(map[string]void)
+	toAccountLabels["DRAFT"] = member
+	toAccountLabels["INBOX"] = member
+	toAccountLabels["SENT"] = member
+	toAccountLabels["TRASH"] = member
+	_, toBeAccounted := toAccountLabels[label]
+	return toBeAccounted
+}
+
+// This function takes a list of labels and count
+// the number of real labels discarding extra labels.
+func toBeAccountedLabelsCount(g *Gmail, labels []string) int64 {
+	// If a label was specified on the command line
+	// then we are interested only on this given label
+	// and we return 1.
+	if len(g.label) != 0 {
+		return 1
+	}
+	// No label was specified so return the relevant labels
+	// count.
+	count := int64(0)
+	for _, label := range labels {
+		if isToBeAccounted(label) {
+			count += 1
+		}
+	}
+	return count
+}
+
+func getLabelsCount(g *Gmail) (int64, error) {
```

Comment:
I'm afraid I don't understand the algorithm here. It seems like we are doing two things

1. Getting the message count per *labels for which we will download messages* (which is either all "toBeAccounted" labels, or the label specified on the command line?). 
2. Getting the *total* count for all toBeAccounted labels. 

And then displaying progress as "(1) divided by (2)". 

I don't understand how this is correct? Isn't (1) the *total* number of matching messages?

Given the code that is updating the progress meter knows its current download counts, I thought the only error in displaying progress was that the *denominator* (i.e. returned from (2)) was wrong (prior to this PR).